platforms/core-configuration/model-core/src/integTest/groovy/org/gradle/internal/extensibility/DeprecatedBooleanPropertyIntegrationTest.groovy

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.extensibility
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class DeprecatedBooleanPropertyIntegrationTest extends AbstractIntegrationSpec {
+    def "does not emit deprecation warning when a decorated class exposes a Boolean property like a field"() {
+        buildFile << """
+            abstract class MyExtension {
+                Boolean property = Boolean.TRUE
+            }
+            def myext = extensions.create("myext", MyExtension)
+            task assertProperty {
+                doLast {
+                    assert myext.property
```

Comment:
Are these tests compatible with CC given they capture the entire extension?

These tests might be a bit more straightforward by doing the assertions outside of a task (and just running help), but if this works for CC no need to change anything

Reply:
> Are these tests compatible with CC given they capture the entire extension?

There is no problem in capturing the extension object, if it is CC-serializable. What is prohibited is to retrieving the extension through `Task.extensions` or Groovy's implicit resolution of such. Here the access goes through the captured variable, so it is safe.